model/Core/Properties/originalNamespaces.md

```diff
@@ -0,0 +1,20 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# namespaces
+
+## Summary
+
+Provides a NamespaceMap representing the NamspaceMap used for the original serialization of an SpdxDocument.
+
+## Description
+
+This field is used for providing the ability to reproduce an original serialization for a given SPDX Document, especially if the serialization format does not natively support namespace mapping.
+
+This purely optional field is not intended for implementing or representing the NameSpaceMap for all serializations where this SPDX document may be found.
```

Comment:
some questions regarding semantics:

- if this is not provided, is the serialization allowed to use any `NameSpaceMap` feature?
  - or more specific: are only the explicitly stated `NameSpaceMap` entries allowed?
- if there is some `NameSpaceMap` like feature of a serialization used, should it be added to the `NameSpaceMap` of the corresponding `SpdxDocument`?
- What are the semantics, if a serialized blob contains not exactly one `SpdxDocument`?

Whether the `optional` works depends somewhat on these questions.

And I see two levels of optional here. The entity minting the SpdxDocument might choose to not add Namespaces and the serializer might choose to not use the NameSpaceMap in the Document. Are both optional? 


maybe not so relevant:
- How would this approach handle potential inconsistent constraint on mappings? There might be keys, that are allowed in some serialization but disallowed in another one.

Reply:
> if this is not provided, is the serialization allowed to use any NameSpaceMap feature?

If your question is related to the native namespace map used by the serialization format (not some property serialized data), there is no restriction whether or not the original namespaces is provided.

> if there is some NameSpaceMap like feature of a serialization used, should it be added to the NameSpaceMap of the corresponding SpdxDocument?

If the creator of the SPDX data wishes to support namespace round tripping, yes.

> What are the semantics, if a serialized blob contains not exactly one SpdxDocument?

Each SpdxDocument object represents one serialization of SPDX data.  If you have a serialization which makes copies of multiple serializations which support the round tripping of namespace data, you can have multiple SPDX documents.  Each SpdxDocument properties can then be used to recreate the original SPDX namespaces.

> And I see two levels of optional here. The entity minting the SpdxDocument might choose to not add Namespaces and the serializer might choose to not use the NameSpaceMap in the Document. Are both optional?

Yes - These are to support recreating the original namespace map.  The creator does not have to support this and can omit the necessary data.  The consumer can ignore the data and create whatever serialization specific namespace map they want.  This is only for the case where the producer wishes to have the namespace map preserved and the consumer would like to recreate it.

> How would this approach handle potential inconsistent constraint on mappings? There might be keys, that are allowed in some serialization but disallowed in another one.

The producer would just record the namespace mapping they used.  If the consumer can't handle the namespace mapping for their chosen serialization format, they would have to handle that in a method specific to the output serialization format.